build.proj

```diff
@@ -102,11 +114,41 @@
     
     <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Restoring all packages...Done." />
   </Target>
+  
+  <!-- Generated project.json's may not exist when the "ProjectJsonFiles"" item group is defined (in dir.props), This target
+       ensures those files are added to the itemgroup. -->
+  <Target Name="AddGeneratedProjectJsons">
+    <ItemGroup>
+      <ProjectJsonFiles Condition="'$(BuildTestsAgainstPackages)' == 'true'" Include="$(GeneratedProjectJsonDir)/**/project.json" />
+    </ItemGroup>
+  </Target>
+  
+  <Import Project="$(ToolsDir)buildagainstpackages.targets" Condition="Exists('$(ToolsDir)buildagainstpackages.targets') and '$(BuildTestsAgainstPackages)' == 'true'" />
+  
+  <Target Name="UpdateVersionsOnTestProjectJson" >
```

Comment:
This is the change to generate  the updated test project.json in front of all other steps to avoid race conditions.